chpl/chpl-service/src/main/java/gov/healthit/chpl/scheduler/job/subscriptions/subjects/formatter/ObservationSubjectFormatterFactory.java

```diff
@@ -35,6 +44,12 @@ public ObservationSubjectFormatter getSubjectFormatter(SubscriptionObservation o
             return criteriaAddedFormatter;
         } else if (lookupUtil.getCertificationCriteriaRemovedSubjectId().equals(observationSubjectId)) {
             return criteriaRemovedFormatter;
+        } else if (lookupUtil.getRwtPlansUrlChangedSubjectId().equals(observationSubjectId)) {
```

Comment:
Just from a code format point of view, should this be a switch / case thing? Or, could we just remove all the "else"s? If we return out of the function on any "true" if statement, then the next ifs won't even be reached

Certainly not a change that's necessary, but am curious if there's interest in some sort of small refactor. Especially as we hope to add more of these down the line, and I _think_ the "if else if else if else" pattern is usually frowned upon. Could be wrong

Reply:
I would be happy to do this, except the "id" that I want to switch on is a Long and Java only allows you to switch on String, enum, byte, short, char, and int. In my research this is because switch is expected to be used on a smallish list of things and if the list is big enough to be of type "long" then you probably wouldn't switch on it. We will never have that many subjects so it doesn't _have_ to be a "long" type, but all of our IDs are either Long or UUID so I don't know about making this one different. I could, I could also refactor and do something with an enum, or convert the Long to a String here and switch on it that way. I dunno. I agree with you that a switch would be better but I feel like it's not a simple change at this point.

Reply:
Fair enough. What about the idea of just removing the `else` portions of all of the `else if` statements? If we return from inside the `if`, then we'll never get to the rest of the checks. I know I'm doing that a lot in the JS stuff where it's effectively short-circuiting the next checks, and I have heard people not like that kind of action, but it makes sense to me. More a passing thought than anything else